api/src/deployment.rs

```diff
@@ -482,7 +488,20 @@ impl DeploymentSystem {
                 // library when the runtime gets around to it.
 
                 let mut lock = deployment.state.write().await;
-                if let DeploymentState::Deployed(DeployedState { so, handle, .. }) = lock.take() {
+                if let DeploymentState::Deployed(DeployedState {
+                    so,
+                    handle,
+                    shutdown_handle,
+                    ..
+                }) = lock.take()
+                {
+                    if let Some(f) = *shutdown_handle {
+                        // TODO: ideally it would receive feedback regarding shutdown
+                        // finalization, and only then we'd move to aborting the thread
+                        // TODO: error handling
+                        f.send(true).ok();
+                    }
+                    // QUESTION: why abort thread instead of stopping the runtime?
```

Comment:
That is a good question, and I believe this to be an error.

Maybe we should not even have the `Handle` in there in the first place but just keep the `Service` inside `DeploymentState` and call its `::shutdown` on signalling for termination.

Reply:
I think having `Service` inside `DeploymentState` would be the most convenient, but also requires bigger refactor, therefore I myself held off from trying it before I showed you guys this one. I'm gonna attempt it next and see how it goes.

Worst case scenario we can pass around the handle to runtime and just shutdown the runtime. I'm sure it will work, but I'm also least excited about that option cause I'm not really certain what happens under the hood when we do that - it would be nice to know this, but that would require a lot more effort (at least for me, unless someone would be interested to help and research this together). Also I'm not sure if shutting down underlying runtime in Rocket vs Axum vs [...] would yield the same effect.

Reply:
[3a78937](https://github.com/getsynth/shuttle/pull/123/commits/3a789379f6fa3adcabd476ae56c54ef8d5fd30c7) resolves it